src/main/java/io/jenkins/plugins/sprp/PipelineSnippetGenerator.java

```diff
@@ -0,0 +1,203 @@
+
+        snippet += "\tsteps {\n";
+        snippet += "\t\t" + addTabs(shellScritp(stage.getScripts()), 2);
+        snippet += "\t}\n";
+
+        if(stage.getFailure() != null
+                || stage.getSuccess() != null
+                || stage.getAlways() != null
+                || (stage.getName().equals("Build") && (archiveArtifacts != null || buildResultPaths != null))
+                || stage.getName().equals("Tests") && testResultPaths != null) {
+            snippet += "\tpost {\n";
+
+            if (stage.getSuccess() != null
+                    || (stage.getName().equals("Build") && (archiveArtifacts != null || buildResultPaths != null))
+                    || stage.getName().equals("Tests") && testResultPaths != null) {
+                snippet += "\t\tsuccess {\n";
+                if (stage.getName().equals("Build")) {
+                    if(archiveArtifacts != null)
+                        snippet += "\t\t\t" + addTabs(getArchiveArtifactsSnippet(archiveArtifacts), 3);
+
+                    if(buildResultPaths != null)
+                        snippet += "\t\t\t" + addTabs(getPublishReportSnippet(buildResultPaths), 3);
+                }
+                if (stage.getName().equals("Tests") && testResultPaths != null) {
+                    snippet += "\t\t\t" + addTabs(getPublishReportSnippet(testResultPaths), 3);
+                }
+                if(stage.getSuccess() != null)
+                    snippet += "\t\t\t" + addTabs(shellScritp(stage.getSuccess()), 3);
+                snippet += "\t\t}\n";
+            }
+            if (stage.getAlways() != null) {
+                snippet += "\t\talways {\n";
+                if(stage.getAlways() != null)
+                    snippet += "\t\t\t" + addTabs(shellScritp(stage.getAlways()), 3);
+                snippet += "\t\t}\n";
+            }
+            if (stage.getFailure() != null) {
+                snippet += "\t\tfailure {\n";
+                snippet += "\t\t\t" + addTabs(shellScritp(stage.getFailure()), 3);
+                snippet += "\t\t}\n";
+            }
+
+            snippet += "\t}\n";
+        }
+        snippet += "}\n";
+
+        return snippet;
+    }
+
+    public String getPublishArtifactStage(ArtifactPublishingConfig config,
+                                          ArrayList<HashMap<String, String>> publishArtifacts){
+        if(config == null)
+            return "";
+
+        StringBuilder snippet = new StringBuilder("stage('Publish Artifact') {\n");
+
+        snippet.append("\tsteps {\n");
+        snippet.append("\t\t" + "withCredentials([file(credentialsId: '").append(config.getCredentialId()).append("', variable: 'FILE')]) {\n");
+
+        for(HashMap<String, String> artifact: publishArtifacts){
+            snippet.append("\t\t\tsh 'scp -i $FILE ").append(artifact.get("from")).append(" ").append(config.getUser()).append("@").append(config.getHost()).append(":").append(artifact.get("to")).append("'\n");
```

Comment:
No need to use manual implementation. https://wiki.jenkins.io/display/JENKINS/SCP+plugin or https://wiki.jenkins.io/display/JENKINS/Publish+Over+SSH+Plugin would do the job assuming they are Pipeline-compatible

Reply:
Oh! I didn't know about this I looked for [SCP Plugin](https://wiki.jenkins.io/display/JENKINS/SCP+plugin) and I didn't use that because it's last development was in 2011. I will try to use this.

Reply:
As discussed, let's keep it as is for now and create a follow-up task